### PR TITLE
Update AWS logger code for AWS SDK 0.12.4

### PR DESCRIPTION
**Pull request by zwass**
The AWS SDK changed how custom HTTP clients are used, and this commit brings
compatibility with the new initialization style.

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by zwass** [comment]
We are still working on Linux provisioning, so this has so far only been tested on OSX. Putting it up for comment.

File: osquery/logger/plugins/aws_util.cpp [hunk 1]
```diff
@@ -8,15 +8,19 @@
  *
  */
 
+#include <mutex>
```
Comment (theopolis) [inline osquery/logger/plugins/aws_util.cpp]: Is this needed for `0.12.4`?

Comment (zwass) [inline osquery/logger/plugins/aws_util.cpp]: This is the header that defines `std::call_once` that we use below. Is there a reason we should be avoiding it?

**Comment by theopolis** [comment]
ok to test

File: osquery/logger/plugins/aws_util.cpp (hunk 1 above)
Comment (theopolis) [inline osquery/logger/plugins/aws_util.cpp]: No reason! Not sure why I didn't make that connection initially. 💃

**Comment by osqueryer** [comment]
Build origin/pr/2140/merge (Job: 2743) failed :-1:

**Comment by theopolis** [comment]
Segfaults on Linux:
```
[----------] 2 tests from AwsUtilTests
[ RUN      ] AwsUtilTests.test_get_credentials
```

Let's try to reproduce locally in the Vagrant targets.

**Comment by theopolis** [comment]
Reproduced on the `ubuntu14

**Comment by zwass** [comment]
We were missing some `initAwsSdk()` calls in the tests, I'm about to push a new commit that should fix things.

**Comment by zwass** [comment]
Note that updates in the AWS SDK cause linker warnings about the use of `tmpnam`, there is an open issue in the SDK repo for this (https://github.com/aws/aws-sdk-cpp/issues/192) and the AWS folks say it's only used for unit tests.

**Comment by osqueryer** [comment]
Build origin/pr/2140/merge (Job: 2746) is good! :+1:

**Comment by zwass** [comment]
Still a few more things I'm checking on, so please don't merge yet.

**Comment by theopolis** [comment]
I saw that issue about `tmpnam`, we should still push them to fix since it will cause failures in our weekly static/dynamic analysis tests. But it is not an issue that will block merging these fixes. :D

**Comment by zwass** [comment]
I manually verified that logs are sent to both kinesis and firehose on OSX, Ubuntu12 and Ubuntu14.

I also ran `make sanitize` on OSX, and both `make sanitize` and `SANITIZE_THREAD=1 make sanitize` on Ubuntu14.

I think this is good to merge now.

re `tmpnam`, I commented to that effect on the issue.

**Pull request merged**